pandas/core/dtypes/common.py

```diff
@@ -1907,18 +1908,18 @@ def _get_dtype_type(arr_or_dtype):
     elif isinstance(arr_or_dtype, DatetimeTZDtype):
         return DatetimeTZDtypeType
     elif isinstance(arr_or_dtype, IntervalDtype):
-        return IntervalDtypeType
+        return Interval
```

Comment:
this is really changing the semantics

Reply:
Which semantics?

Reply:
all the other dtypes return the meta class
now you are returning the actual class (and not even the type)

this is not right

Reply:
This is what we have been doing for all EAs.

Now in practice, I don't if we ever *rely* on the value, though?

Reply:
I agree that am not sure this is actually used. But If you are chaning, try changing for all the dtypes? (or don't change). that's what I mean by semantics, we neeed to be consistent.

Reply:
From the ExtensionArray docs

```
        """The scalar type for the array, e.g. ``int``

        It's expected ``ExtensionArray[item]`` returns an instance
        of ``ExtensionDtype.type`` for scalar ``item``.
        """
 ```

Period is the scalar type for arrays with PeriodDtype, in the sense that `isinstance(periodarray[0], Period)` is true. What's the issue?

> now you are returning the actual class (and not even the type)


Classes are types.

Reply:
> But If you are chaning, try changing for all the dtypes? (or don't change)

We discussed earlier change CategoricalDtype.type to be object, but IIRC that was rejected (I don't recall the specifics).

Happy to change `DatetimeTZDtypeType` (to `Timestamp`).

Reply:
can you see if we have tests that hit *any* of this? (if not pls add). happy to change to either way, but should just change all.

Reply:
tests/dtypes/test_common.py::test__get_dtype_type hits this. Otherwise, the only path to here seems to be `_get_dtype_or_type`.

Reply:
And even then, for EA's it is actually never used in `_get_dtype_or_type` (I mean, there is no check function relying on that value, only for numpy dtypes)